langchain_tools_demo/main.py

```diff
@@ -95,6 +93,20 @@ async def chat_handler(request: Request, prompt: str = Body(embed=True)):
         raise HTTPException(status_code=500, detail=f"Error invoking agent: {err}")
 
 
+@app.post("/reset")
+async def reset(request: Request):
+    """Reset agent"""
+    global user_agents
+    uuid = request.session["uuid"]
+    if uuid in user_agents.keys():
+        user_agent = user_agents[uuid]
+        await user_agent.client.close()
+        del user_agent.agent
```

Comment:
why we are deleting just the agent here?

Reply:
added deleting agent client